modules/alex-n/src/main/java/com/hillel/elementary/javageeks/cellular_automaton/Main.java

```diff
@@ -0,0 +1,135 @@
+package com.hillel.elementary.javageeks.cellular_automaton;
+
+import java.util.Scanner;
+
+import static com.hillel.elementary.javageeks.cellular_automaton.MatrixProcessor.*;
+
+public class Main {
+    public static void main(String[] args){
+
+        int rows = 0;
+        int columns = 0;
+        Scanner scanner = new Scanner(System.in);
+        boolean initializingData = true;
+
+        while (initializingData) {
+            System.out.printf("Please enter number of columns: ");
+            if (scanner.hasNextInt()) {
+                columns = scanner.nextInt();
+            } else {
+                System.out.printf("wrong data. Exit");
+                System.exit(0);
+            }
+            System.out.printf("Please enter number of rows: ");
+            if (scanner.hasNextInt()) {
+                rows = scanner.nextInt();
+            } else {
+                System.out.printf("wrong data. Exit");
+                System.exit(0);
+            }
+            if (columns <= 0 || rows <= 0){
+                System.out.println("Error. Can't sat matrix: " + columns + " x " + rows + ". Dimensions must be grater than 0. Try again.");
+                initializingData = true;
+            }else{
+                initializingData = false;
+            }
+        }
+
+        int[][] matrix = new int[columns][rows];
+
+        matrix = fillMatrix(matrix);
+        System.out.println("Initialized matrix:");
+        printMatrix(matrix);
+
+        matrix = oneLifeCycle(matrix);
+        System.out.println("Matrix after one life cycle:");
+        printMatrix(matrix);
+    }
+}
+
+
+class MatrixProcessor{
+
+    //fill matrix with random 0 or 1
+    public static int[][] fillMatrix(int[][] matrix){
+
+        for (int i = 0; i < matrix[0].length; i++) {
+            for (int j = 0; j < matrix.length; j++) {
+                matrix[j][i] = (int) Math.round(Math.random());
```

Comment:
Method "nextInt()" of the "Random" class may be used too. For example: Random.nextInt(2)
https://www.tutorialspoint.com/java/util/random_nextint_inc_exc.htm